plugin/trino-lakehouse/src/main/java/io/trino/plugin/lakehouse/LakehouseMetadata.java

```diff
@@ -0,0 +1,1023 @@
+            return null;
+        }
+        if (isIcebergTable(table)) {
+            return icebergMetadata.getTableHandle(session, tableName, startVersion, endVersion);
+        }
+        if (isDeltaLakeTable(table)) {
+            return deltaMetadata.getTableHandle(session, tableName, startVersion, endVersion);
+        }
+        if (isHudiTable(table)) {
+            return hudiMetadata.getTableHandle(session, tableName, startVersion, endVersion);
+        }
+        return hiveMetadata.getTableHandle(session, tableName, startVersion, endVersion);
+    }
+
+    @Override
+    public Optional<ConnectorTableExecuteHandle> getTableHandleForExecute(ConnectorSession session, ConnectorAccessControl accessControl, ConnectorTableHandle tableHandle, String procedureName, Map<String, Object> executeProperties, RetryMode retryMode)
+    {
+        return forHandle(tableHandle).getTableHandleForExecute(session, accessControl, tableHandle, procedureName, executeProperties, retryMode);
+    }
+
+    @Override
+    public Optional<ConnectorTableLayout> getLayoutForTableExecute(ConnectorSession session, ConnectorTableExecuteHandle tableExecuteHandle)
+    {
+        return forHandle(tableExecuteHandle).getLayoutForTableExecute(session, tableExecuteHandle);
+    }
+
+    @Override
+    public BeginTableExecuteResult<ConnectorTableExecuteHandle, ConnectorTableHandle> beginTableExecute(ConnectorSession session, ConnectorTableExecuteHandle tableExecuteHandle, ConnectorTableHandle updatedSourceTableHandle)
+    {
+        return forHandle(tableExecuteHandle).beginTableExecute(session, tableExecuteHandle, updatedSourceTableHandle);
+    }
+
+    @Override
+    public void finishTableExecute(ConnectorSession session, ConnectorTableExecuteHandle tableExecuteHandle, Collection<Slice> fragments, List<Object> tableExecuteState)
+    {
+        forHandle(tableExecuteHandle).finishTableExecute(session, tableExecuteHandle, fragments, tableExecuteState);
+    }
+
+    @Override
+    public void executeTableExecute(ConnectorSession session, ConnectorTableExecuteHandle tableExecuteHandle)
+    {
+        forHandle(tableExecuteHandle).executeTableExecute(session, tableExecuteHandle);
+    }
+
+    @Override
+    public Optional<SystemTable> getSystemTable(ConnectorSession session, SchemaTableName tableName)
+    {
+        return hiveMetadata.getSystemTable(session, tableName)
+                .or(() -> icebergMetadata.getSystemTable(session, tableName))
+                .or(() -> deltaMetadata.getSystemTable(session, tableName))
+                .or(() -> hudiMetadata.getSystemTable(session, tableName));
+    }
+
+    @Override
+    public Optional<ConnectorTableHandle> applyPartitioning(ConnectorSession session, ConnectorTableHandle tableHandle, Optional<ConnectorPartitioningHandle> partitioningHandle, List<ColumnHandle> columns)
+    {
+        return forHandle(tableHandle).applyPartitioning(session, tableHandle, partitioningHandle, columns);
```

Comment:
Should we verify that `tableHandle` and `partitioningHandle` are for the same format like in `getCommonPartitioningHandle` as well or is that not expected to happen because this API is applied only during table scanning (if I understand the new API correctly)?

Reply:
Looking at `getCommonPartitioningHandle` again, the verification seems potentially incorrect, as it might be called for e.g., an Iceberg and Hive table, in which case they should be treated as having incompatible partitioning.

For this method, the partitioning always comes from the table, so it shouldn't be possible for them to mismatch.